interfaces/Matlab_Toolbox_Revamp/Class/Kinetics.m

```diff
@@ -0,0 +1,590 @@
+            %    for all reactions. If this function is called without
+            %    argument, a bar graph is produced. 
+            
+            checklib;
+            nr = kin.nReactions;
+            xx = zeros(1, nr);
+            pt = libpointer('doublePtr', xx);
+            calllib(ct, 'kin_getFwdRateOfProgress', kin.kin_id, nr, pt);
+            q = pt.Value;
+%             if nargout == 0
+%                 figure
+%                 set(gcf, 'Name', 'Rates of Progress')
+%                 bar(q)
+%                 xlabel('Reaction Number')
+%                 ylabel('Forward Rate of Progress [kmol/m^3]')
+%                 title('Forward Rates of Progress')
+%             end
+        end
+        
+        function q = rop_r(kin)
+            % Get the reverse rates of progress for all reactions.
+            %
+            % return:
+            %    Returns a column vector of the reverse rates of progress
+            %    for all reactions. If this function is called without
+            %    argument, a bar graph is produced. 
+            
+            checklib;
+            nr = kin.nReactions;
+            xx = zeros(1, nr);
+            pt = libpointer('doublePtr', xx);
+            calllib(ct, 'kin_getRevRateOfProgress', kin.kin_id, nr, pt);
+            q = pt.Value;
+%             if nargout == 0
+%                 figure
+%                 set(gcf, 'Name', 'Rates of Progress')
+%                 bar(q)
+%                 xlabel('Reaction Number')
+%                 ylabel('Reverse Rate of Progress [kmol/m^3]')
+%                 title('Reverse Rates of Progress')
+%             end
+        end
+        
+        function q = rop(kin)
+            % Get the forward and reverse rates of progress.
+            %
+            % return:
+            %    Returns an I x 2 array of reaction rates of progress,
+            %    where I is the number of reactions. The first column
+            %    contains the forward rates of progress, and the second
+            %    column the reverse rates. If this function is called
+            %    without arguments, a bar graph is produced. 
+            
+            checklib;
+            f = rop_f(kin);
+            r = rop_r(kin);
+            q = [f, r];
+%             if nargout == 0
+%                 figure
+%                 set(gcf, 'Name', 'Rates of Progress')
+%                 bar(q)
+%                 xlabel('Reaction Number')
+%                 ylabel('Rate of Progress [kmol/m^3]')
+%                 title('Rates of Progress')
+%                 legend('Forward', 'Reverse')
+%             end
+        end        
+       
+        function q = rop_net(kin)
+            % Get the net rates of progress for all reactions.
+            %
+            % return:
+            %    Returns a column vector of the net rates of progress
+            %    for all reactions. If this function is called without
+            %    argument, a bar graph is produced. 
```

Comment:
Since the graph code is commented out, this docstring is not correct. But I think you should delete the graph code and update all these docstrings anyways.